File: repl/adrianus/cpp/test_parser.cpp
#include "parser.h"
#include "utils.cpp"

void test_parse_program() {
    Parser parser;
    parser.Load("let a = 1; let add = fn(a,b) { 1+2+3; } ; werty; ; 1 + 2 - 3 * 4; add(a, 747); if (3 > 1) { 1001; } else { 1002; } let a = true; let b = false;");
    //parser.Load("while (3 < 10) { 1; }");
    Ad_AST_Program program;
    parser.ParseProgram(program);
    print_ast_nodes(&program, 0);
}

void test_funtion_literal() {
    Parser parser;
    parser.Load("let a = fn(x) { return x+1; }");
    //parser.Load("while (3 < 10) { 1; }");
    Ad_AST_Program program;
    parser.ParseProgram(program);
    print_ast_nodes(&program, 0);
}

void run_parser_tests() {
    test_parse_program();
}

File: repl/adrianus/python/test_parser.py
from repl import Repl
from parser import Parser
from ast import ASTProgram
from utils import print_ast_nodes


def test_parser():
    source = "let a = 1; let add = fn(a,b) { 1+2+3; } ; werty; ; 1 + 2 - 3 * 4; add(a, 747); if (3 > 1) { 1001; } else { 1002; } let a = true; let b = false;"
    parser = Parser()
    program = ASTProgram()
    parser.reset(source=source)
    program.reset()
    parser.build_program_statements(program)
    print_ast_nodes(program, 0)

def test_parser2():
    source = "let a = 1;"
    parser = Parser()
    program = ASTProgram()
    parser.reset(source=source)
    program.reset()
    parser.build_program_statements(program)
    print_ast_nodes(program, 0)
